### PR TITLE
CNI shouldn't pass trustdomain down

### PR DESCRIPTION
**Pull request by bleggett**
Fixes: https://github.com/istio/istio/issues/51182

We don't need CNI to do this, and CNI doesn't have this context anyway.

EDIT: zt side is in.

### CODE REVIEW

File: pkg/zdsapi/zds.proto [hunk 1]
```diff
@@ -33,7 +33,6 @@ message WorkloadInfo {
   string name = 1;
   string namespace = 2;
   string service_account = 3;
-  string trust_domain = 4;
```
Comment (howardjohn) [inline pkg/zdsapi/zds.proto:36]: probably need to reserve it?

**Review by howardjohn — APPROVED** [review]
LGTM but this might need to change ztunnel first to stop comparing the TD?

**Comment by bleggett** [comment]
Back compat:

1. The zt proto changes will be backported to 1.22, so that 1.22 zt can talk to 1.23 CNI
2. The CNI changes should not need to be backported, as 1.22 CNI talking to 1.23 zt should be fine b/c of protobuf compat guarantees.

**Comment by bleggett** [comment]
/test integ-ambient
/test unit-tests

Not sure what's up here..

**Comment by bleggett** [comment]
Oh duh. Obviously, the ambient integ tests will fail until zt is updated in this repo.

**Pull request merged**